test/distributed/_composable/test_compose.py

```diff
@@ -145,6 +147,33 @@ def test_checkpoint_fsdp_submodules_use_reentrant(self):
     def test_checkpoint_fsdp_submodules_non_reentrant(self):
         self._test_checkpoint_fsdp_submodules(False)
 
+    @skip_if_lt_x_gpu(2)
+    def test_fully_shard_replicate_correct_replicate_params(self):
+        model = CompositeParamModel(device=torch.device("cuda"))
+        # Shard Linears within UnitModule
+        fully_shard(model.u1, policy=ModuleWrapPolicy({nn.Linear}))
+        fully_shard(model.u2, policy=ModuleWrapPolicy({nn.Linear}))
+        # replicate the rest
+        replicate(model)
+        # Run fwd + bwd to initialize DDP
```

Comment:
Curious: what does it mean to initialize DDP? Do you have to run backward?

Reply:
In replicate, DDP is only initialized in the first forward. You don't have to run backward, will clarify that in the comment